tests/lib/managers/files-test.js

```diff
@@ -2952,4 +2952,164 @@ describe('Files', function() {
 
 	});
 
+	describe('getRepresentationInfo()', function() {
+		var TEST_REPRESENTATION = '[png?dimensions=1024x1024]';
+		var expectedRepresentationParam = { headers: { 'x-rep-hints': TEST_REPRESENTATION }, qs: { fields: 'representations' } };
+		var expectedRepresentationAllParams = { headers: { 'x-rep-hints': TEST_REPRESENTATION, set_content_disposition_type: 'inline', set_content_disposition_filename: 'New_File_Name.png' }, qs: { fields: 'representations' } };
+		var placeholderUrl = '.../{+asset_path}';
+		var placeholderUrlWithoutAssetPath = '.../2.png';
+		var representationOptions = {asset_path: '1.png'};
+		var representationAllOptions = {asset_path: '1.png', set_content_disposition_type: 'inline', set_content_disposition_filename: 'New_File_Name.png'};
+
+		it('should make GET request to get file representation when called', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [{
+							content: {
+								url_template: placeholderUrl
+							}
+						}]
+					}
+				}
+			};
+			sandbox.mock(boxClientFake).expects('get').withArgs('/files/1234', expectedRepresentationParam)
+				.returns(Promise.resolve(response));
+			files.getRepresentationInfo(FILE_ID, TEST_REPRESENTATION, representationOptions);
+		});
+
+		it('should make GET request to get file representation when called with all optional headers', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [{
+							content: {
+								url_template: placeholderUrl
+							}
+						}]
+					}
+				}
+			};
+			sandbox.mock(boxClientFake).expects('get').withArgs('/files/1234', expectedRepresentationAllParams)
+				.returns(Promise.resolve(response));
+			files.getRepresentationInfo(FILE_ID, TEST_REPRESENTATION, representationAllOptions);
+		});
+
+		it('should call callback with the representation when a 200 response is returned', function(done) {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [
+							{
+								content: {
+									url_template: placeholderUrl,
+									asset_url: placeholderUrlWithoutAssetPath
```

Comment:
This isn't quite right; this value isn't returned in the API response.  It's best to make the test cases line up with how the API actually responds to avoid introducing weird bugs.